group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorShard.java

```diff
@@ -309,6 +311,21 @@ public CoordinatorResult<OffsetCommitResponseData, Record> commitOffset(
         return offsetMetadataManager.commitOffset(context, request);
     }
 
+    /**
+     * Handles a ListGroups request.
+     *
+     * @param context The request context.
+     * @param request The ListGroups request.
+     * @return A Result containing the ListGroupsResponseData response
+     */
+    public ListGroupsResponseData listGroups(
+        RequestContext context,
+        ListGroupsRequestData request,
+        long committedOffset
+    ) throws ApiException {
```

Comment:
I wonder if we should simplify the signature of this method. How about taking a list of states and returning a list of ListedGroup? The full request and the context are not really necessary in my opinion in this case.

Reply:
I tried to simplify the function signature based on my own understanding. The reason why the function return value here is consistent with the current one is that when an exception is thrown, the final result with an error code can be directly returned. Otherwise, the Runtime exception can only be thrown upward in completeExceptionaly. (List<ListsGroup> cannot pass in an error code). Then when handling the exception, you need to parse the cause of the RuntimeException, which I'm not sure is appropriate. Do you have any suggestions?